lib/rules/max-lines-per-function.js

```diff
@@ -157,6 +157,33 @@ module.exports = {
             return false;
         }
 
+        /**
+         * Search for the closest identifer for the node function.
+         * @param {ASTNode} node Node function
+         * @param {ASTNode} parent Parent node of the function
+         * @returns {ASTNode} The function identifer node or the function body if none exists
+         */
+        function reportedNode(node, parent) {
```

Comment:
If we accept this change, we could use [`astUtils.getFunctionHeadLoc`](https://github.com/eslint/eslint/blob/d75d3c68ad8c98828aaa522b87ec267ab2dcb002/lib/rules/utils/ast-utils.js#L1765) utlity.